samples/apps/copilot-chat-app/SKWebApi/Config/ConfigExtensions.cs

```diff
@@ -20,7 +21,14 @@ public static IHostBuilder ConfigureAppSettings(this IHostBuilder host)
             builder.AddJsonFile($"appsettings.{environment}.json", true, true);
             builder.AddEnvironmentVariables();
             builder.AddUserSecrets(Assembly.GetExecutingAssembly(), optional: true, reloadOnChange: true);
-            // For settings from Key Vault, see https://learn.microsoft.com/en-us/aspnet/core/security/key-vault-configuration?view=aspnetcore-7.0
+            // For settings from Key Vault, see https://learn.microsoft.com/en-us/aspnet/core/security/key-vault-configuration?view=aspnetcore-8.0
+            string? keyVaultName = ctx.Configuration["KeyVaultName"];
+            if (!string.IsNullOrEmpty(keyVaultName))
+            {
+                builder.AddAzureKeyVault(
+                    new Uri($"https://{keyVaultName}.vault.azure.net/"),
+                    new DefaultAzureCredential());
```

Comment:
DefaultAzureCredential is a *beast*. Please add a reference to its documentation to keep the insanity meter from filling up. https://learn.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet

Reply:
Sure